pkg/kubelet/kubelet.go

```diff
@@ -796,11 +809,13 @@ func (kl *Kubelet) validateNodeIP() error {
 	return fmt.Errorf("Node IP: %q not found in the host's network interfaces", kl.nodeIP.String())
 }
 
+// allSourcesReady returns whether all seen pod sources are ready.
 func (kl *Kubelet) allSourcesReady() bool {
 	// Make a copy of the sourcesSeen list because it's not thread-safe.
 	return kl.sourcesReady(sets.NewString(kl.sourcesSeen.List()...))
 }
 
+// addSource adds a new pod source.
```

Comment:
I think add source adds a new source to the list of sources seen, this is fine until we get #24810 which adds thread-safety...